utils/s3/client.cc

```diff
@@ -89,6 +90,19 @@ future<> ignore_reply(const http::reply& rep, input_stream<char>&& in_) {
 client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global_factory gf, private_tag, std::unique_ptr<aws::retry_strategy> rs)
         : _host(std::move(host))
         , _cfg(std::move(cfg))
+        , _creds_invalidation_timer([this] {
+            s3l.info("Invalidating creds");
+            _credentials = {};
+        })
+        , _creds_update_timer([this] {
+            (void)[this]->future<> {
+                s3l.info("Update creds in the background");
+                _credentials = co_await _creds_provider_chain.get_aws_credentials();
```

Comment:
Probably need some verbosity in case get_aws_credentials() throws and fails to update creds. Presumably "synchronous" update will tell us what the problem is.

Another question -- if this get_aws_credentials() fails, does it make sense to re-arm the times for now() + SomeMinutes in a hope that it was a temporary glitch?

Reply:
> Probably need some verbosity in case get_aws_credentials() throws and fails to update creds. Presumably "synchronous" update will tell us what the problem is.


> Another question -- if this get_aws_credentials() fails, does it make sense to re-arm the times for now() + SomeMinutes in a hope that it was a temporary glitch?

Not really, [later on](https://github.com/scylladb/scylladb/pull/21933/commits/c7cf0e26391ca707651e617e45592a68863e8900) retryable client will be introduced to SMS/EC2Metadata provider, so if we failed for 15 retries, then IMO we should give up